docs/reference/egress.md

```diff
@@ -0,0 +1,128 @@
+# Egress Proxy
+
+**Disclaimer**: Egress features are probably not feature
+complete. Please create [Github Issues](https://github.com/zalando/skipper/issues)
+to show your ideas about this topic.
+
+The picture below shows an authentication use case with Bearer token
+injection, to show the egress traffic flow:
+![egress traffic flow](../img/egress-traffic-flow.svg)
+
+Skipper has some features, which are egress specific. Some features,
+for example [dropRequestHeader](filters.md#droprequestheader) or
+[ratelimit](filters.md#ratelimit), might also be used, but are not
+listed here:
+
+* circuit breaker filters
+   * [consecutiveBreaker](filters.md#consecutivebreaker)
+   * [rateBreaker](filters.md#ratebreaker)
+   * [disableBreaker](filters.md#disablebreaker)
+* [bearerinjector](filters.md#bearerinjector) filter, that injects tokens for an app
+* The secrets module that does
+   * automated secrets rotation read from files used by `bearerinjector filter`
+   * dynamic secrets lookup used by `bearerinjector filter`
+   * encryption and decryption used by [OpenID Connect filters](filters.md#oauthoidcuserinfo)
+
+## Secrets Module
+
+**Disclaimer**: the specified features might be changed to make use
+  cases work in the future.
+
+### Automated Secrets rotation
+
+Secrets are read from files. Files can be rewritten by third party
+tools to integrate whatever provider you want.
+In Kubernetes you can write
+[Secrets](https://kubernetes.io/docs/concepts/configuration/secret/)
+with an API and read them using a rotated, mounted files from skipper
+for example.
+
+To specify files or directories to find secrets, you can use
+`-credentials-paths` command line flag. Filenames are used to define
+the name of the secret, which will be used as a lookup key.
+
+Files need to be created before skipper started, and as of today
+skipper doesn't find new files automatically. This might change in the
+future.
+
+To change the default update interval, which defaults to `10m`, you
+can use the `-credentials-update-interval` command line flag.
+
+#### Example bearer injection
+
+Create file `/tmp/secrets/mytoken`, that contains `mytoken`:
+
+```
+mkdir /tmp/secrets; echo mytoken >/tmp/secrets/mytoken`.
+```
+
+start fake service
+
+```
+nc -l 8080
+```
+
+start skipper forward proxy
+
+```
+skipper -inline-routes='Host("host1") -> bearerinjector("sec1") -> "http://127.0.0.1:8080/"' -credentials-paths=/tmp/secrets -credentials-update-interval=10s
```

Comment:
I wouldn't call this yet a forward proxy, because it is not. For that we would need the dynamic backend set from the incoming scheme and host. Let's just call it a proxy here, and explain it in a separate paragraph that this behavior is missing.

Reply:
I see that it's metnioned below, so probably it's just enough not to call it a forward proxy.